src/org/gk/slicing/SlicingEngine.java

```diff
@@ -239,6 +239,65 @@ public void slice() throws Exception {
         setStableIdReleased();
     }
     
+    /**
+     * Iterate through EntitySet instances and populate each instance's
+     * disease and compartment attributes.
+     * 
+     * <h3>Purpose</h3>
+     * Consider an EntitySet composed of all drugs in its hasMember slot, 
+     * each of which may have different set of diseases.
+     * We want to collect all diseases together into the disease slot of this EntitySet.
+     * 
+     * <h3>Recursion</h3>
+     * An EntitySet may contain another EntitySet. So this should be done recursively.
+     * The same thing should be done for the compartment slot. 
+     * 
+     * <h3>Application Set</h3>
+     * In the actual implementation, we should apply the above rule to all EntitySets,
+     * not necessary for EntitySet having drugs only.
+     * 
+     * @param GKInstance, String...
+     * @throws Exception
+     */
+    private void populateEntitySet(GKInstance instance, String... attributeNames) throws Exception {
+    	Set<GKInstance> attributes = InstanceUtilities.getContainedInstances(instance, 
+    			ReactomeJavaConstants.hasMember,
+    			ReactomeJavaConstants.hasCandidate);
+    	// If EntitySet has a "hasMember" or "hasCandidate" attribute then recursively iterate over them.
+    	if (attributes != null && attributes.size() > 0) {
+    		for (GKInstance attribute: attributes) {
+    			populateEntitySet(attribute);
```

Comment:
The above is a recursive method. There is no need to call this again.

Reply:
Fixed in [1c3f71d](https://github.com/reactome/CuratorTool/pull/20/commits/1c3f71df3a6bf4ee5e125378de7c9fba5bf91cc1).